Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2041,6 +2041,18 @@ void WebPageProxy::loadRequestWithNavigationShared(Ref<WebProcessProxy>&& proces
     auto transaction = pageLoadState->transaction();
 
     auto url = request.url();
+#if PLATFORM(COCOA)
+    if (!url.isNull() && !url.isValid() && WTF::linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::ConvertsInvalidURLsToNull)) {
```

Comment:
I prefer these kind of compound boolean clauses to be put in a convenience variable (or two) to improve readability. Especially when there are multiple negations in the clause like this one. Please consider something like `auto urlValidAndNotNull = !url.isNull() && !url.isValid();`.